pyproject.toml

```diff
@@ -1,55 +1,72 @@
 [build-system]
-requires = ["poetry_core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
 
-[tool.poetry]
+[project]
 name = "matrix-nio"
-version = "0.25.0"
+version = "0.25.0+nexy7574.dependencies"
 description = "A Python Matrix client library, designed according to sans I/O principles."
 authors = [
-    "Damir Jelić <poljar@termina.org.uk>",
-    "Paarth Shah <mail@shahpaarth.com>"
+    { name = "Damir Jelić", email = "poljar@termina.org.uk" },
+    { name = "Paarth Shah", email = "mail@shahpaarth.com" }
 ]
-license = "ISC"
+license = { file = "LICENSE.md" }
 readme = "README.md"
-repository = "https://github.com/poljar/matrix-nio"
-documentation = "https://matrix-nio.readthedocs.io/en/latest/"
-packages = [
-    {include = "nio"}
+requires-python = ">=3.8.0"
+dependencies = [
+    "aiohttp~=3.10",
+    "aiofiles~=24.1",
+    "h11~=0.14",
+    "h2~=4.0",
+    "jsonschema~=4.14",
+    "unpaddedbase64~=2.1",
+    "pycryptodome~=3.10",
+    "aiohttp-socks~=0.8"
 ]
 
-[tool.poetry.dependencies]
-python = "^3.8.0"
-aiohttp = "^3.10.3"
-aiofiles = "^24.1.0"
-h11 = "^0.14.0"
-h2 = "^4.0.0"
-jsonschema = "^4.14.0"
-unpaddedbase64 = "^2.1.0"
-pycryptodome = "^3.10.1"
-python-olm = { version = "^3.2.16", optional = true }
-peewee = { version = "^3.14.4", optional = true }
-cachetools = { version = "^5.3.3", optional = true }
-atomicwrites = { version = "^1.4.0", optional = true }
-aiohttp-socks = "^0.8.4"
+[tool.setuptools.packages]
+find = { include = ["nio", "nio.*", "nio.*.*"] }
+# TODO: an `src/` directory should be used for setuptools. This is an incredibly horrible hack for the time being.
 
-[tool.poetry.extras]
-e2e = ["python-olm", "peewee", "cachetools", "atomicwrites"]
+[project.optional-dependencies]
+e2e = [
+    "python-olm~=3.2",
+    "peewee~=3.14",
+    "cachetools~=5.3",
+    "atomicwrites~=1.4"
+]
+
+# TODO: Should e2e dependencies be included in test/test dependencies? or just rely on the e2e extra being installed?
```

Comment:
One thing that I know that would work: this causes the dependencies specified in other optional groups to be included automatically (and no version specifier is needed since `matrix-nio` is a self-reference to the project itself
dev = [
    "matrix-nio[e2e]",
    ...
]

Reply:
I think this actually references the published version of the project - `.` references the current project in pyproject, but I've got a feeling this might cause a dependency loop. I'll try it anyway

Reply:
Huh, using the pep508 name *does* work. I didn't know that, neat.